src/binary/uint.rs

```diff
@@ -73,6 +69,45 @@ impl UInt {
     }
 }
 
+/// The result of the [`magnitude_to_bytes`] function. The fields are private;
+/// users must go through `as_ref()` to see the slice that ignores leading
+/// zeros.
+///
+/// This type exists to avoid heap allocation.
+pub struct U64Magnitude {
+    magnitude_bytes: [u8; mem::size_of::<u64>()],
+    first_occupied_byte: usize,
+}
+
+impl AsRef<[u8]> for U64Magnitude {
+    fn as_ref(&self) -> &[u8] {
+        &self.magnitude_bytes[self.first_occupied_byte..]
+    }
+}
+
+/// Returns the magnitude as big-endian bytes.
+///
+/// ```
+/// use ion_rs::binary::uint;
+/// use std::convert::AsRef;
+///
+/// let repr = uint::magnitude_to_bytes(5u64);
+/// assert_eq!(&[0x05], repr.as_ref());
+/// ```
```

Comment:
Doc test is nice, but some unit boundary tests for the slices that you produce would be good.  They are probably covered by integration, but I'd still say its worth it.

Reply:
I added one more test to the doctest to show two bytes. Otherwise, I played around with the implementation a bit (e.g. changed the division by 8) and coverage is excellent by tests that are already in this file.